java/vector/src/main/java/org/apache/arrow/vector/complex/AbstractStructVector.java

```diff
@@ -40,11 +40,64 @@
 public abstract class AbstractStructVector extends AbstractContainerVector {
   private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(AbstractContainerVector.class);
 
+  private static final ConflictPolicy DEFAULT_CONFLICT_POLICY;
   // Maintains a map with key as field name and value is the vector itself
-  private final MapWithOrdinal<String, FieldVector> vectors = new MapWithOrdinal<>();
+  private final PromotableMultiMapWithOrdinal<String, FieldVector> vectors;
+  protected final boolean allowConflictPolicyChanges;
+  private ConflictPolicy conflictPolicy;
 
-  protected AbstractStructVector(String name, BufferAllocator allocator, CallBack callBack) {
+
+  static {
+    String conflictPolicyStr = System.getProperty("arrow.struct.conflict.policy", "CONFLICT_REPLACE");
```

Comment:
if you move the enum above this block, are you able to access the CONFLICT_REPLACE enum directly, to use it instead oa constant here?

Reply:
fixed